src/test/scala/is/hail/expr/ir/AggregatorsSuite.scala

```diff
@@ -383,6 +383,35 @@ class AggregatorsSuite {
       FastSeq(43L, 36L)
     )
 
+  private[this] def assertRequiredArraySumEvalsTo[T: HailRep](
```

Comment:
@catoverdrive I tried to test the flexibility I added in the aggregators (allowing for required types), but this new test is failing when type checking in the IR because requiredness is stripped (although not deeply removed) from the seqop args. If this test isn't valid because we don't allow required types, then the changes I made to the aggregators above shouldn't be needed if all requiredness is removed. But then we get the error with the BGEN type being required... I'm not sure what the correct behavior should be.

Reply:
Not all requiredness is stripped--only the top level requiredness. what happens if you change line 390 to `val typ = TArray(hailType[T].setRequired(true))`?

Reply:
(that would basically be the same change you made in the parser here:
https://github.com/hail-is/hail/pull/4233/files/de24c9f7872931126c674d45537dc4782f6b62cd#diff-aa0ed8949d28819102d347a40600a8eeR521)